packages/sync-actions/src/utils/action-map-custom.js

```diff
@@ -26,21 +26,32 @@ const extractTypeFields = (diffedFields, nextFields) =>
     : nextFields
 const extractFieldValue = (newFields, fieldName) => newFields[fieldName]
 
-export default function actionsMapCustom(diff, newObj, oldObj) {
+export default function actionsMapCustom(
+  diff,
+  newObj,
+  oldObj,
+  customActions = {}
+) {
   const actions = []
+  const priceId = oldObj.id || undefined
+  if (customActions.setCustomType)
+    Actions.setCustomType = customActions.setCustomType
+  if (customActions.setCustomField)
+    Actions.setCustomField = customActions.setCustomField
   if (!diff.custom) return actions
   if (hasSingleCustomFieldChanged(diff)) {
     // If custom is not defined on the new or old category
     const custom = diffpatcher.getDeltaValue(diff.custom, oldObj)
-    actions.push({ action: Actions.setCustomType, ...custom })
+    actions.push({ action: Actions.setCustomType, priceId, ...custom })
```

Comment:
This works because `priceId` is `undefined` if the `newObj`/`oldObj` aren't prices? I know that the assignment above `const priceId = oldObj.id || undefined` already existed before your changes, but to me it looks a bit dangerous. A lot of our types have an `id`... Whot do you think about it @MarkusAzer ?

Reply:
Thats totally right, should i pass it as optional params like this maybe
`export default function actionsMapCustom(
  diff,
  newObj,
  oldObj,
  custom = {
    customActions = {},
    priceId
  }
)`

what do you think?!

Reply:
I think it's a good idea, but would like to hear opinions from the others too. 
CC: @daern91 @jherey @danrleyt

Reply:
Yeah, I can't think of anything else, I like your idea @MarkusAzer. This way we can guarantee that has a value.

Reply:
I think your approach works, let's go with this. We can make updates later if we find a better way.